python/tank/platform/validation.py

```diff
@@ -156,18 +164,50 @@ def validate_and_return_frameworks(descriptor, environment):
         # (new) - {"name": "tk-framework-qtwidgets", "version": "v1.x.x"}
         # 
         # The new syntax requires a floating version number, meaning that 
-        # the framework instance defined in the environment needs to be on the form  
+        # the framework instance defined in the environment needs to be of the form  
         # 
         # frameworks:
         #   tk-framework-qtwidgets_v1.x.x:
         #     location: {name: tk-framework-qtwidgets, type: app_store, version: v1.3.34}
         #
         desired_fw_instance = "%s_%s" % (name, version)
-        for fw_instance_name in fw_instances:
+        min_version_satisfied = True
+
+        for fw_instance_name, fw_desc in fw_descriptors.iteritems():
+
+            # We've found a matching framework.
             if fw_instance_name == desired_fw_instance:
-                found = True
-                required_fw_instance_names.append(fw_instance_name)
-                break
+                # Now we need to see if there's a minimum required version
+                # of the framework. If we got a v2.x.x framework that flattens
+                # out to v2.1.0 and we have a minimum required version of v2.1.5,
+                # then we haven't actually found a compatible framework.
+                #
+                # If the descriptor for the framework doesn't contain a concrete
+                # version number (example: a "dev" descriptor won't have one), we
+                # simply skip the minimum-required version check. There's nothing
+                # we can do to confirm, so it's best to trust the config. It's
+                # likely that anyone using a non-app_store descriptor understands
+                # the caveats that come with them.
+                fw_version = fw_desc.version
```

Comment:
Sticking with just a warning in the case where the comparison can't be made for a technical reason.